gremlin-go/driver/graphBinary.go

```diff
@@ -583,6 +578,28 @@ func pathReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return p, nil
 }
 
+// Format: Same as List.
+// Mostly similar to listWriter and listReader with small changes
+func setWriter(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+	slice, err := value.(Set).SetToSlice()
+	if err != nil {
+		return nil, err
+	}
+	b, err := listWriter(slice, buffer, typeSerializer)
+	if err != nil {
+		return nil, err
+	}
+	return b, err
```

Comment:
listWriter will return nil, err if err != nil, so let's short circuit this.
```suggestion
	return listWriter(slice, buffer, typeSerializer)
```